AppinfoJsonFileType.js

```diff
@@ -0,0 +1,168 @@
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var path = require("path");
+var log4js = require("log4js");
+var AppinfoJsonFile = require("./AppinfoJsonFile.js");
+var logger = log4js.getLogger("loctool.plugin.AppinfoJsonFileType");
+
+var AppinfoJsonFileType = function(project) {
+    this.type = "json";
+    this.resourceType = "json";
+    this.project = project;
+    this.extensions = [".json"];
+
+    /* in order to match proper xliff
+    * datatype value is using to create reskey
+    */
+    var dataTypeMap = {
+        "webos-web":"javascript",
+        "webos-qml": "x-qml",
+        "webos-cpp": "cpp",
+        "webos-c": "cpp"
+    }
+
+    this.datatype = dataTypeMap[project.options.projectType] || "javascript";
+
+    this.API = project.getAPI();
+    this.extracted = this.API.newTranslationSet(project.getSourceLocale());
+    this.newres = this.API.newTranslationSet(project.getSourceLocale());
+    this.pseudo = this.API.newTranslationSet(project.getSourceLocale());
+};
+
+/**
+ * Return true if the given path is a java file and is handled
+ * by the current file type.
+ *
+ * @param {String} pathName path to the file being questions
+ * @returns {boolean} true if the path is a java file, or false
+ * otherwise
+ */
+AppinfoJsonFileType.prototype.handles = function(pathName) {
+    logger.debug("AppinfoJsonFileTyp handles " + pathName + "?");
+    if (!pathName) return false;
+    
+    return (pathName === "appinfo.json") ? true : false
+};
+
+AppinfoJsonFileType.prototype.name = function() {
+    return "Appinfo Json File Type";
+};
+
+/**
+ * Write out the aggregated resources for this file type. In
+ * some cases, the string are written out to a common resource
+ * file, and in other cases, to a type-specific resource file.
+ * In yet other cases, nothing is written out, as the each of
+ * the files themselves are localized individually, so there
+ * are no aggregated strings.
+ */
+AppinfoJsonFileType.prototype.write = function(translations, locales) {
+    // templates are localized individually, so we don't have to
+    // write out the resources
+};
+
+AppinfoJsonFileType.prototype.newFile = function(path) {
+    return new AppinfoJsonFile({
+        project: this.project,
+        pathName: path,
+        type: this
+    });
+};
+
+AppinfoJsonFileType.prototype.getDataType = function() {
+    return this.datatype;
+};
+
+AppinfoJsonFileType.prototype.getResourceTypes = function() {
+    return {};
+};
+
+/**
+ * Return the name of the node module that implements the resource file type, or
+ * the path to a javascript file that implements the resource filetype.
+ * @returns {Function|undefined} node module name or path, or undefined if this file type does not
+ * need resource files
+ */
+AppinfoJsonFileType.prototype.getResourceFileType = function() {
+    return AppinfoJsonFileType;
```

Comment:
Don't need a resource file type for appinfo.json files because we don't generate resource files